packages/@aws-cdk/aws-codeartifact/lib/repository.ts

```diff
@@ -0,0 +1,404 @@
+
+  /**
+     * The domain repository owner
+     * Often, equivalent to the account id.
+     * @attribute
+     */
+  readonly repositoryDomainOwner: string;
+
+  /**
+     * The domain the repository belongs to
+     * @attribute
+     */
+  readonly repositoryDomainName: string;
+
+  /**
+     * Grants the given IAM identity permissions to read from the repository
+     */
+  grantRead(identity: iam.IGrantable): iam.Grant;
+
+  /**
+     * Grants the given IAM identity permissions to write from the repository
+     */
+  grantWrite(identity: iam.IGrantable): iam.Grant;
+
+  /**
+     * Grants the given IAM identity permissions to read/write from the repository
+     */
+  grantReadWrite(identity: iam.IGrantable): iam.Grant;
+
+  /**
+     * Defines a CloudWatch event rule which triggers for repository events. Use
+     * `rule.addEventPattern(pattern)` to specify a filter.
+     */
+  onEvent(id: string, options?: events.OnEventOptions): events.Rule;
+
+  /**
+     * Defines a CloudWatch event rule which triggers when a "CodeArtifact Package
+     *  Version State Change" event occurs.
+     */
+  onPackageVersionStateChange(id: string, options?: events.OnEventOptions): events.Rule;
+}
+
+/**
+ * Properties for a new CodeArtifact repository
+ */
+export interface RepositoryProps {
+  /**
+     * Name the repository
+     */
+  readonly repositoryName: string,
+  /**
+     * A text description of the repository.
+     * @default ''
+     */
+  readonly description?: string,
+  /**
+     * The name of the domain that contains the repository.
+     */
+  readonly domain: IDomain,
+  /**
+     * Upstream repositories for the repository
+     * @see https://docs.aws.amazon.com/codeartifact/latest/ug/repos-upstream.html
+     * @default None
+     */
+  readonly upstreams?: IRepository[],
+  /**
+     * An array of external connections associated with the repository.
+     * @default None
+     * @see https://docs.aws.amazon.com/codeartifact/latest/ug/external-connection.html#adding-an-external-connection
+     */
+  readonly externalConnections?: e.ExternalConnection[],
+
+  /**
+     * Principal to associate allow access to the repository
+     * @default AccountRootPrincipal
+     */
+  readonly principal?: iam.IPrincipal
+
+  /**
+   * Principal to associate allow access to the repository
+   * @default Read/Write
+   */
+  readonly policyDocument?: iam.PolicyDocument
+}
+
+/**
+ * Properties for a new CodeArtifact repository policy restricting package manipulation
+ */
+export interface PolicyRepositoryPackage {
+  /**
+     * Package format
+     */
+  readonly packageFormat: string,
+  /**
+     * Package namespace
+     */
+  readonly packageNamespace: string,
+  /**
+     * Package name
+     */
+  readonly packageName: string,
+}
+
+/**
+ * Either a new or imported Repository
+ */
+export abstract class RepositoryBase extends Resource implements IRepository {
+
+  /** @attribute */
+  public abstract readonly repositoryArn: string = '';
```

Comment:
Don't need to re-label these as attributes, or initialize them.
```suggestion
  public abstract readonly repositoryArn: string;
```